### ISSUE TITLE
[Feature Request] Add extension to relative import id

### ISSUE THREAD

**Issue by loynoir** [evanw/esbuild#1563]
### Feature Request
Add extension `.mjs` to relative import id,

when specific `--out-extension:.js=.mjs`

### Repeoduce
```ts
import deps from "./deps"
console.log(deps)
```

```sh
esbuild \
    index.ts \
    --out-extension:.js=.mjs \
    --outdir=dist
```

### Actual
```ts
import deps from "./deps";
console.log(deps);
```

```txt
$ node dist/index.mjs
Error [ERR_MODULE_NOT_FOUND]: Cannot find module 'path/to/dist/deps' imported from path/to/dist/index.mjs
```

### Expected
```ts
import deps from "./deps.mjs";
console.log(deps);
```

```txt
$ node dist/index.mjs
42
```

**Comment by hyrious** [comment]
tsc doesn't append .mjs too.
see #622

**Comment by evanw** [comment]
Yes, TypeScript works this way by design, and therefore esbuild's TypeScript support also works this way. If you would like for Microsoft to change how TypeScript works you should ask them, not me. I don't work at Microsoft and don't have any control over their roadmap. I believe https://github.com/microsoft/TypeScript/issues/27957 is a relevant issue to follow.

The workarounds for this situation are described here: https://github.com/evanw/esbuild/issues/622#issuecomment-752915822. Either use `.js` and `"type": "module"` or write your own esbuild plugin to implement the behavior you want.

**Comment by loynoir** [comment]
@evanw 

As the bottleneck here is relative import , what do you think of https://github.com/evanw/esbuild/issues/1564 ?

Allow CLI accessing some out-of-box external preset.

In this case, add an out-of-box preset mark all non-relative import id as external.

So, the output will has no relative import, both typescript and node will happy.

**Comment by stefanvonderkrone** [comment]
if someone is interrested in my solution:
https://github.com/evanw/esbuild/issues/622#issuecomment-1370603292

**Comment by shellscape** [comment]
I think it's completely reasonable for a compiler to add a file extension as a transformation. Compilers make all kinds of adjustments for other nuanced mechanics of intent. At the very least, a plugin should solve this. And for what it's worth, the TS team has been repeatedly chided by the community for the decision. Their decision making isn't exactly pure.
